---
title: "Kahle exploration"
output: html_notebook
---

## What does this do?

This script is for exploring the kahle et al data concerning splicing in cancer. We want  to find out a few things:

1) Unique splice junctions found in tumours versus normal
 1.1. Look for EGFRvIII --> glioblastoma
 1.2 Claudin 18.2 (although I recall this was tricky last time...) --> pancreatic / gastric
 1.3 CD44v6 (although the pipeline looks dead) --> Breast?
 
2) Looks for their peptide evidence and where we can find the elution profiles of these sorts of data

## Libraries and packages


```{r}

library(tidyr)
library(data.table)
library(dplyr)
library(ggplot2)

```


## Loading data


Ok let's try the first file

```{r}
fnam <- "../input/tss_complexity_counts.whitelisted.G0.01.globsum20.filtLib.conf3_neojunctions_corrected.tsv.gz"


tmp <- fread(fnam, header = F)

tmp %>% head

```

Looks like we have sample information, gene, and chromosome positions. Presumably these correspond to junction points of exons. In the paper they say they call "â‰ˆ251,000" such neojunctions. They seem to arrive at this point by essentially filtering TCGA non-GENCODE junctions againts GTEx and normal junctions. If the table we just loaded covers these solely (i.e. it has already been filtered) we should see this by uniquifying the concatenation of V2 and V3

```{r}
gene_coord_combos <- paste0(tmp$V2, "_", tmp$V3)

gene_coord_combos %>% unique %>% length


```

Indeed that is what we see. Ok let's familiarize ourselves with how they got to this number. If they filtered AGAINST gencode or even GTEx it could be that they've filtered out things that do have physiological presence like claudin 18.2

Ok we have something here: "For a ranking of neojunctions, we sorted all EEJs with an increased specificity towards tumor samples requiring a minimum number of spliced alignments across the EEJ per sample to count it as expressed (tumor: 10 spliced reads, normals: 3 spliced reads, GTEx: 2 spliced reads). Further, we removed all junctions that were present in more than 1% of GTEx or TCGA normal samples or had a higher mean expression in TCGA normals compared to TCGA tumor samples (within the same cancer type). We then ranked all EEJs by predominant occurrence in tumor samples based on Fisher's exact test. To aggregate over multiple events in a gene, we show only the event with the strongest effect."


Ok so here we essentially see that they have a rough definition of what tumor-specific is. They say "we need at least 10 spliced reads in say its in a tumor sample, 3 if its a matched normal, 2 if it is a GTEx. Furthermore, we exclude it if its in 1% of GTEx samples.

Now... this means that the claudin 18.2 event (that is, inclusion of an alternate exon 1) might be filtered out if it is counted at that... more or less arbitrary cutoff.

We can try to download all splice junction events called by Kahle et al and try to fish them out


## Exploring the Kahle et al splice event calls

Kahle et al. give, on their webpage (see https://gdc.cancer.gov/about-data/publications/PanCanAtlas-Splicing-2018), different files. Some are large, the ones I have downloaded are mostly labelled with "confirmed" which I take to mean that they think they are confirmed by mass spectrometry (or at least they think they are). I cannot imagine that they were though: CPTAC at the time did not have coverage of all the tumor types and neojunctions covered by the paper. I think it is rather, in the manuscript, the arbitrary threshold of 20 reads that is being used to call something "confirmed".

Let's load the files and explore them. Note that the supplementary files table was generated running the aforementioned website text through chat gpt to generate a table I could operate on.

```{r}

call_dir <- "../input/individual_calls/"

downloaded_files <- list.files(call_dir)

manifest <-  "../input/20250820_KahleSupplementaryFiles.xlsx"


manifest_tab <- openxlsx::read.xlsx(manifest)

manifest_tab

```

We'll add a label to see what we have actually downloaded

```{r}

manifest_tab$Downloaded <- FALSE

manifest_tab$Downloaded[manifest_tab$File.Name %in% downloaded_files] <- TRUE


```


Now let's select for just the "confirmed"


```{r}

coords <- grep(pattern = "merge_graphs", fixed = T, manifest_tab$File.Name)


mini_manifest_tab <- manifest_tab[coords,]


## and then we restrict to "confirmed" which might still be bullshit

mini_manifest_tab <- mini_manifest_tab %>% 
  filter(grepl(pattern = "confirmed", x = File.Name))
```


Ok great, we have the files. Let's select some and explore them. They come in two flavours: txt.gz or gff3. Let's first ask perplexity.ai what the difference is and how we can load them into R.

Perplexity points out that the gff3 files are specifically for genomic features and their annotation. The files are consistently smaller than the txt.gz, so they may function as accompanying information

